commands.js

```diff
@@ -5,8 +5,201 @@
+ *   If a room is not specified, it will default to the current one.
+ *   If it has a falsy value, the check won't be attached to any room.
+ *   In addition to running the checks from this.canTalk(), it also
+ *   checks to see if the message has any banned words, is too long,
+ *   or was just sent by the user. Returns the filtered message, or a
+ *   falsy value if the user can't speak.
+ *
+ *   Should usually be near the top of the command, like:
+ *     target = this.canTalk(target);
+ *     if (!target) return false;
+ *
+ * this.parse(message, inNamespace)
+ *   Runs the message as if the user had typed it in.
+ *
+ *   Mostly useful for giving help messages, like for commands that
+ *   require a target:
+ *     if (!target) return this.parse('/help msg');
+ *
+ *   If `inNamespace` is true, then the message is parsed in that
+ *   corresponding namespace:
+ *     // command msg is in namespace test. (ie. /test msg)
+ *     this.parse('/help', true); // is parsed as if the user said
+ *                                // '/test help'
+ *
+ *   After 10 levels of recursion (calling this.parse from a command
+ *   called by this.parse from a command called by this.parse etc)
+ *   we will assume it's a bug in your command and error out.
+ *
+ * this.targetUserOrSelf(target, exactName)
+ *   If target is blank, returns the user that sent the message.
+ *   Otherwise, returns the user with the username in target, or
+ *   a falsy value if no user with that username exists.
+ *   By default, this will track users across name changes. However,
+ *   if exactName is true, it will enforce exact matches.
+ *
+ * this.getLastIdOf(user)
+ *   Returns the last userid of an specified user.
+ *
+ * this.splitTarget(target, exactName)
+ *   Splits a target in the form "user, message" into its
+ *   constituent parts. Returns message, and sets this.targetUser to
+ *   the user, and this.targetUsername to the username.
+ *   By default, this will track users across name changes. However,
+ *   if exactName is true, it will enforce exact matches.
+ *
+ *   Remember to check if this.targetUser exists before going further.
+ *
+ * Unless otherwise specified, these functions will return undefined,
+ * so you can return this.sendReply or something to send a reply and
+ * stop the command there.
+ *
+ * A /help entry can be added for a command by adding an entry to any
+ * command file in the form "<command>help:", where <command> is the
+ * primary function name, followed by an array or function.
+ * To add a separate entry /help entry for an alias, use "<alias>help:"
+ * and that help entry will be returned instead of the one for the
+ * primary function name.
+ * If the help entry is an array each element of the array will be
+ * sent as a new line. If the help entry is a function the help
+ * command will execute that function.
+ * Help commands are able to be left both for namespace objects and
+ * within namespace objects for commands within them.
```

Comment:
If somebody could read my documentation for how to add help entries to make sure it's adequately clear, I'd appreciate it.